modules/system/classes/extensions/ModuleManager.php

```
@@ -0,0 +1,283 @@
+
+        return $this;
+    }
+
+    /**
+     * @return array<string, WinterExtension>
+     */
+    public function list(): array
+    {
+        return array_merge(...array_map(fn($key) => [$key => $this->get($key)], Config::get('cms.loadModules', [])));
+    }
+
+    public function create(string $extension): WinterExtension
+    {
+        throw new ApplicationException('Support for creating extensions needs implementing');
+    }
+
+    public function install(WinterExtension|ExtensionSource|string $extension): WinterExtension
+    {
+        // Get the module code from input and then update the module
+        if (!($code = $this->resolveIdentifier($extension))) {
+            throw new ApplicationException('Unable to update module: ' . $code);
+        }
+
+        // Force a refresh of the module
+        $this->refresh($code);
+
+        // Return an instance of the module
+        return $this->get($code);
+    }
+
+    public function enable(WinterExtension|string $extension, string|bool $flag = self::DISABLED_BY_USER): mixed
+    {
+        // TODO: Implement enable() method.
+        throw new ApplicationException('Support for enabling modules needs implementing');
+    }
+
+    public function disable(WinterExtension|string $extension, string|bool $flag = self::DISABLED_BY_USER): mixed
+    {
+        // TODO: Implement disable() method.
+        throw new ApplicationException('Support for disabling modules needs implementing');
+    }
+
+    /**
+     * @throws ApplicationException
+     */
+    public function update(WinterExtension|string|null $extension = null, bool $migrationsOnly = false): ?bool
+    {
+        $modules = $this->getModuleList($extension);
+
+        $firstUp = UpdateManager::instance()->isSystemSetup();
+
+        if ($firstUp) {
+            $this->repository->createRepository();
+            $this->output->info('Migration table created');
+        }
+
+        if (!$migrationsOnly && !Config::get('cms.disableCoreUpdates')) {
```

We should probably include an info or warning message if `cms.disableCoreUpdates` is set; I used to use it on every site to prevent the backend from updating the core when I was actually managing them with composer so I could see it causing issues if we silently ignore them using that setting.

---

see: https://github.com/wintercms/winter/pull/1259/commits/4e339308a4fbce24803adee44c9fbb84018c9146